src/pages/FullArticlePage.jsx

```diff
@@ -3,13 +3,38 @@ import { useParams } from 'react-router-dom'
 import {getArticleById} from '../utils/api'
 import CommentsPopUp from '../components/CommentsPopUp'
 import Popup from '../components/Popup'
+import { VoteByArticleId } from '../utils/api'
 
 
 const FullArticlePage = () =>{
+    const [article, setArticle] = useState({})
+    const [votes, setVotes] = useState(0)
     const [PopupComments, setPopupComments]= useState(false)
-    const [article, setArticle] = useState("")
     const {article_id} =useParams()
 
+    const handleVote = (change) =>{
+        if(change === +1){
+        let voteData = {inc_votes:1}
+        setVotes((currentVotes)=>{return currentVotes+1})
+        VoteByArticleId(article_id, voteData)
+        .then(()=>{
+            console.log('Upvote successful')
+        })
+        .catch((err)=>{
+            console.error(err)
+        })}else{ 
+            let voteData = {inc_votes:-1}
+        setVotes((currentVotes)=>{return currentVotes-1})
+        VoteByArticleId(article_id,voteData)
+        .then(()=>{
+            console.log('Downvote successful')
+        })
+        .catch((err)=>{
+            console.error(err)
+        })
+        }
```

Comment:
You could get this to work without the if/else, if you can get the change parameter to just be a value of 1 or -1, then you could replace where you have hardcoded 1 in inc_votes, currentVotes + 1 etc with the change value and not need the if else